File: Leetcode 0225 Implement Stack using Queues.go
// 20250729
type MyStack struct {
    q1 []int // main
    q2 []int
}


func Constructor() MyStack {
    return MyStack{}
}


func (this *MyStack) Push(x int)  {
    this.q2 = append(this.q2, x)
    for len(this.q1) > 0 {
        this.q2 = append(this.q2, this.q1[0])
        this.q1 = this.q1[1:]
    }
    this.q1, this.q2 = this.q2, this.q1
}


func (this *MyStack) Pop() int {
    a := this.q1[0]
    this.q1 = this.q1[1:]
    return a
}


func (this *MyStack) Top() int {
    return this.q1[0]
}


func (this *MyStack) Empty() bool {
    return len(this.q1) == 0
}


/**
 * Your MyStack object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Push(x);
 * param_2 := obj.Pop();
 * param_3 := obj.Top();
 * param_4 := obj.Empty();
 */

File: Leetcode 0225 Implement Stack using Queues.py
# 20250729
class MyStack:

    def __init__(self):
        self.l = []

    def push(self, x: int) -> None:
        self.l = [x] + self.l

    def pop(self) -> int:
        t = self.l[0] 
        self.l = self.l[1:]
        return t

    def top(self) -> int:
        return self.l[0]

    def empty(self) -> bool:
        return len(self.l) == 0


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()
